backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -468,6 +476,24 @@ public Response updateEndpoint(@Context SecurityContext sec,
             String processorId = dbProperties.getExtras().get(OB_PROCESSOR_ID);
             String processorName = dbProperties.getExtras().get(OB_PROCESSOR_NAME);
 
+            /*
+             * Check that the processor is in an actionable state
+             * before updating it in RHOSE. This will achieve two
+             * goals:
+             *
+             * 1. Avoid returning the RHOSE error directly to the
+             * user.
+             * 2. Avoid unnecessarily calling RHOSE.
+             *
+             * Check https://issues.redhat.com/browse/RHCLOUD-22281
+             * for more information.
+             */
+            if (previousEndpointStatus != EndpointStatus.READY && previousEndpointStatus != EndpointStatus.FAILED) {
+                throw new InternalServerErrorException(
+                        "the processor associated with the endpoint is not ready to be modified"
+                );
+            }
```

Comment:
Can we move that check to line 462 and then avoid storing the status in `previousEndpointStatus`?

Reply:
Regarding the HTTP error that will be returned, I'd return a 400 with the same message instead of a 500.

Reply:
Sure thing! Done!